.github/workflows/pre-commit.yml

```
@@ -0,0 +1,19 @@
+name: Run pre-commit
+
+on:
+  workflow_call:
+
+jobs:
+  run-pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v5
```

Need to checkout the repo first
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```
CI is failing on this repo because of this